### PR TITLE
feat: Finish integrations page

### PR DESCRIPTION
**Pull request by ivasilov**
This PR fixes several issues to finish the migration of the integration pages:
- Make the code block prettier.
- Add the `Admonition` component because it's used in some integration pages.
- Some pages might not have a `Learn` link, so don't show a section for if it's NULL.
- Move the OAuth and Integration (renamed to Marketplace) pages.
- Remove all integrations pages since they've been moved to the DB.

### CODE REVIEW

File: apps/docs/components/Navigation/NavigationMenu/NavigationMenu.constants.ts [hunk 1]
```diff
@@ -1060,6 +1061,30 @@ export const platform: NavMenuConstant = {
         { name: 'Production Checklist', url: '/guides/platform/going-into-prod' },
       ],
     },
+    {
```
Comment (hieuwu) [inline apps/docs/components/Navigation/NavigationMenu/NavigationMenu.constants.ts:1064]: Just wonder. In line 1032 you added a line break. If it is needed, please also add a line break here.
```suggestion
    
    {
```

Comment (hieuwu) [reply apps/docs/components/Navigation/NavigationMenu/NavigationMenu.constants.ts:1063]: Great work! 👍

**Comment by kiwicopple** [comment]
image

looks like we removed the "Integrations" from the sidebar - perhaps we leave it there, but it can pop out to the partners site (in a new window)?

File: apps/docs/components/Navigation/NavigationMenu/NavigationMenu.constants.ts [hunk 2]
```diff
@@ -1060,6 +1053,30 @@ export const platform: NavMenuConstant = {
...
+      name: 'Integrations',
+      url: undefined,
+      items: [
+        {
+          name: 'Supabase Marketplace',
+          url: '/guides/platform/marketplace',
+        },
+      ],
+    },
```
Comment (kiwicopple) [inline apps/docs/components/Navigation/NavigationMenu/NavigationMenu.constants.ts:1079]:
```suggestion
    {
      name: 'Integrations',
      url: undefined,
      items: [
        {
          name: 'Integrations Marketplace',
          url: '/guides/platform/marketplace',
        },
        {
          name: 'Publish an OAuth App',
          url: '/guides/platform/oauth-apps/publish-an-oauth-app',
        },
        {
          name: 'Authorize an OAuth App',
          url: '/guides/platform/oauth-apps/authorize-an-oauth-app',
        },
      ],
    },
```

I'd just group all of these into one section - they all "fit" together pretty nicely

**Review by gregnr — COMMENTED** [review]
Nice one @ivasilov! Just a few things:
- Do we have a 1-to-1 mapping between old doc integration pages and new pages on partner site? We should add [redirects](https://github.com/supabase/supabase/blob/master/apps/www/lib/redirects.js) for each one (or maybe wildcard) so that any external links don't break
- We should do a quick search through the docs to see where we link to doc integrations and manually change to the new links (eg. [prisma reference in database docs](https://github.com/supabase/supabase/blob/bc0a41618e3dc930f23bd997407527d76720244a/apps/docs/pages/guides/database/connecting-to-postgres.mdx#L41))
- Did you end up getting CodeHike to render properly on the partner pages? We've recently been making a switch away from the `CodeBlock` component in favour of code hike. Might be worth referencing how we do something similar [here](https://github.com/supabase/supabase/blob/93db839f885d93b3cb55a19a1ac342f178bacba4/apps/docs/pages/guides/ai/python/%5Bslug%5D.tsx#L129-L142). Happy to help!

File: apps/www/components/Admonition.tsx [hunk 3]
```diff
@@ -0,0 +1,50 @@
+import { PropsWithChildren } from 'react'
+import { IconAlertTriangle, IconHelpCircle, IconInfo } from 'ui'
+
+export interface AdmonitionProps {
+  type: 'note' | 'tip' | 'info' | 'caution' | 'danger'
+  label?: string
+}
+
+const Admonition = ({ type = 'note', label, children }: PropsWithChildren<AdmonitionProps>) => {
```
[the remark this answers is not in the document]
Comment (ivasilov) [reply apps/www/components/Admonition.tsx]: Great suggestion! I've moved the `Admonition` component to the `ui` package and updated all its imports.

File: apps/docs/components/Navigation/NavigationMenu/NavigationMenu.constants.ts (hunk 2 above)
Comment (ivasilov) [reply apps/docs/components/Navigation/NavigationMenu/NavigationMenu.constants.ts]: Fixed.

**Comment by ivasilov** [comment]
> Nice one @ivasilov! Just a few things:
> 
>     * Do we have a 1-to-1 mapping between old doc integration pages and new pages on partner site? We should add [redirects](https://github.com/supabase/supabase/blob/master/apps/www/lib/redirects.js) for each one (or maybe wildcard) so that any external links don't break
> 
>     * We should do a quick search through the docs to see where we link to doc integrations and manually change to the new links (eg. [prisma reference in database docs](https://github.com/supabase/supabase/blob/bc0a41618e3dc930f23bd997407527d76720244a/apps/docs/pages/guides/database/connecting-to-postgres.mdx#L41))
> 
>     * Did you end up getting CodeHike to render properly on the partner pages? We've recently been making a switch away from the `CodeBlock` component in favour of code hike. Might be worth referencing how we do something similar [here](https://github.com/supabase/supabase/blob/93db839f885d93b3cb55a19a1ac342f178bacba4/apps/docs/pages/guides/ai/python/%5Bslug%5D.tsx#L129-L142). Happy to help!

Hey @gregnr, I did all your suggestions! Just a warning, the PR has gotten bigger :). 

I added redirects in `redirects.js` but I'm not sure if they work. They can probably only be tested on production because of their unique URL setup, right?

I updated all links in the docs and blog posts I could find. Let me know if I missed anything. I also switched the `pre` element to using `CodeHike`.

Let me know if I missed anything.

**Review by gregnr — APPROVED** [review]
@ivasilov Nice work 👍 looking good.

You can test redirects on staging via the `www` site:
https://zone-www-dot-com-git-feat-finish-integrations-page-supabase.vercel.app/docs/guides/integrations/auth0

Looks like its working.

Heads up, might have an image issue on the auth0 page:
image

**Comment by ivasilov** [comment]
> @ivasilov Nice work 👍 looking good.
> 
> You can test redirects on staging via the `www` site: https://zone-www-dot-com-git-feat-finish-integrations-page-supabase.vercel.app/docs/guides/integrations/auth0
> 
> Looks like its working.
> 
> Heads up, might have an image issue on the auth0 page: image

The image issues are part of the dynamic content (it's loaded via DB/Storage). I'll go over each page and fix that once I have this PR deployed, no worries.

**Pull request merged**